### ISSUE TITLE
MLP-Mixer: An all-MLP Architecture for Vision

### ISSUE THREAD

**Issue by mei28** [mei28/research_paper#67]
---
id: "mlp-mixer-an-all-mlp-architecture-for-vision"
aliases:
  - "MLP-Mixer: An all-MLP Architecture for Vision"
tags:
  - "TSUNDOKU"
---

# MLP-Mixer: An all-MLP Architecture for Vision

## 一言で言うと

MLPを用いて，ViTのように画像を分割するパッチで処理．単純な行列演算のみで，事前学習ありではSOTAと同等の性能を出した．

### 論文リンク

https://arxiv.org/abs/2105.01601

### 著者/所属機関

- Ilya Tolstikhin, Neil Houlsby, Alexander Kolesnikov, Lucas Beyer, Xiaohua Zhai, Thomas Unterthiner, Jessica Yung, Andreas Steiner, Daniel Keysers, Jakob Uszkoreit, Mario Lucic, Alexey Dosovitskiy

### 投稿日付(yyyy/MM/dd)

NeurIPS2021

## 先行研究と比べてどこがすごい？

- CNNs, self attentionを使わない

## 技術・手法のキモはどこ？

- token-mixing, channel-mixingを使う

## コメント

- 図の色が綺麗.図の見た目も参考にしたい.
- 最適化手法，正規化手法がとてもモリモリ．なんでこうしたのか試行錯誤した様子知りたい．
- token-mixingとchannel-mixingを逆にしたらどうなんだろう

## 次はなに読む？

--- 

- Mixerは一般的な行列演算のみで処理
- 画像をpatch(token)としてpatch x channelに変換を行う 
- Mixerは極端なケースでは1x1のCNNのようになる．
- 100M以下の大規模データセットを用いて事前学習した時は，SOTAと同等の性能を出した．
- 10M以下のデータセットでは，さまざまな正規化手法を用いることで，良い成果を達成することができた．ただViTには及ばない．

- 画像処理では大きく分けて3つの特徴量をmixする方法がとられる．
1. おなじ空間情報をつかう
    - CNNでは1x1に該当 
    - MLP-blockはここに該当
2. 異なる空間情報をつかう
    - CNNではNxNの畳み込みとpoolingで実装
3. 両方を採用
    - 大きなカーネルは両方の性質をもつ
    - Transformer系ではここに該当
MLP-Mixerではchannel-mixingによって1を，token-mixingによって2を実装する．
- 入力画像を重複がないようにSこのpatchに分割 ->  `X \in R^{S x C}, S:= HW/P**2`
- toke-mixing: patchを一列に並べ，転置したX^Tに対して写像 -> R^S -> R^S
- channel-mixing: patchを一列に並べ，転置したX^Tに対して写像 -> R^C -> R^C
- それぞれのmixing層では2つのFC層をもつ
![image](https://file.notion.so/f/s/58799001-fc64-4b0e-9fc7-1749dcfe8e4a/Untitled.png?id=197b2cbc-4721-472c-b8f6-1762f189035a&table=block&spaceId=5acdf6c2-4b40-44fb-9e7f-bd517f633b6e&expirationTimestamp=1681636877050&signature=bMfpwrI6Qr5g1pH4u12OGVrSXAZph-5tAdRp3ZwTNJ8&downloadName=Untitled.png)
- sigmaはelement-wise nonlinearity関数.例えばGELU
    - GELUはTransformer系でよく使われている．
- Mixerではサイズはずっと固定している．これはCNNのピラミッド構造とは違う．
- Mixerではskip-connection, layer-normalizeationを使う．ViTと違い，postion embeddingsは使わない.これはtoken-mixingが入力の順序に対して敏感であるから．

- 中規模〜大規模データセットでの事前学習で分類問題を解く
- 評価指標として
    1. 下流タスクのAccuracy
    2. 事前学習にかかるコスト
    3. 推論にかかる時間
- 分類タスクとしてILSVRC2021"ImageNet"(1.3Mimages, 1kclasses), CIFAR-10/100(50kimages, 10/100class), Oxford-IIIT Pets(3.7kimages, 36classes), Oxford Flowers-102(2k images, 102class)を使う．ベンチマークデータセットとしてVTAB-1kを用いる
- ILSVRC2021, ImageNet-21Kを使って，事前学習を行う．
- 事前学習時の解像度は224, Adam, linear learning rate warmupを10kごとでlinear decay, batch size=4096, weight decay, gradient clipping at global norm1
- imageの事前処理としてcrop, forizontal flipをする．またデータ拡張と正規化手法として，RandmAugment, mixup, dropout, stochastic depth
    - これらの最適化手法は全てtimmによるもの
- fine-tuningでは，momentumSGD, batchsize=512, gradiwnt clipping at global norm1, cosine learning rate schedule with a linear warmup.
- weight decayは行わない．
- 評価しようとして，1. TPU-v3の事前学習時の計算コスト（FLOPs, 効率性，Throughput） 2. top-1のaccuracyで見る．

- 各モデルの帰納バイアスがどうなのか見る
1. パッチをランダムにシャッフルに，各パッチに対して，同じpixelの順序入れ替えをする．
1. 全体でpixelの順序入れ替えをする. pixelの入れ替えは全ての画像を通して同じ． 
- 予測通り，Mixerは順番入れ替えに影響を受けていない．ResNetは強い帰納バイアスを受けている．
![Untitled](https://file.notion.so/f/s/2f546b64-eb0e-4916-b97f-cbda03664862/Untitled.png?id=347258e7-4f2c-4e0f-94ed-d9a6a003e4df&table=block&spaceId=5acdf6c2-4b40-44fb-9e7f-bd517f633b6e&expirationTimestamp=1681636913826&signature=RoADZdWgEtgOMduBOnEjWw7JbLwv4RItBK2f7H52Z_4&downloadName=Untitled.png)

---
